buildSrc/src/main/kotlin/publish-conventions.gradle.kts

```
@@ -0,0 +1,75 @@
+plugins {
+    `maven-publish`
+    signing
+}
+
+// empty xxx-javadoc.jar
+val javadocJar by tasks.register<Jar>("javadocJar") {
+    archiveClassifier.set("javadoc")
+}
+
+publishing {
+    repositories {
+        maven {
+            url = mavenRepositoryUri()
+            credentials {
+                username = project.getSensitiveProperty("libs.sonatype.user")
+                password = project.getSensitiveProperty("libs.sonatype.password")
+            }
+        }
+    }
+
+    publications.withType<MavenPublication>().configureEach {
+        pom {
+            name = project.name
+            description = "AtomicFU utilities"
+            url = "https://github.com/Kotlin/kotlinx.atomicfu"
+
+            licenses {
+                license {
+                    name = "The Apache Software License, Version 2.0"
+                    url = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+                    distribution = "repo"
+                }
+            }
+
+            developers {
+                developer {
+                    id = "JetBrains"
+                    name = "JetBrains Team"
+                    organization = "JetBrains"
+                    organizationUrl = "https://www.jetbrains.com"
+                }
+            }
+
+            scm {
+                url = "https://github.com/Kotlin/kotlinx.atomicfu"
+            }
+        }
+
+
+        signPublicationIfKeyPresent(project, this)
+        // add empty javadocs
+        if (name != "kotlinMultiplatform") { // The root module gets the JVM's javadoc JAR
+            artifact(javadocJar)
+        }
```

It'd be nice to tidy up this config in a later PR.

```suggestion
        // TODO https://github.com/Kotlin/kotlinx-atomicfu/issues/421
        //      Figure out if there's an alternative to string-based config
        //      (if the name changes this config will silently stop working).
        //      Maybe the Javadoc JAR can always be added?
        // add empty javadocs
        
        if (name != "kotlinMultiplatform") { // The root module gets the JVM's javadoc JAR
            artifact(javadocJar)
        }
```